### ISSUE TITLE
False positive for top-level header

### ISSUE THREAD

**Issue by stur86** [cpplint/cpplint#375]
The check for `build/include_subdir` only checks that a subdirectory is present at all; it does not check that it's complete or correct, and it does not allow for top level headers that simply don't have a subdirectory. E.g. if the file structure was

```
src
|
|-- main.cpp
|
|-- utils.hpp
```

and we had a `main.cpp` like

```c++

#include "utils.hpp"

int main() {
}
```

this would raise an error, even though everything is correct.

**Comment by A-lexisL** [comment]
Did you encounter this error? `Include the directory when naming header files  [build/include_subdir]`? 
I have some subdirectories each corresponding to different subprojects, and including header files in these subprojects will raise this error even though I run cpplint just under the very subdirectory. 
If I include the header files like `./feature1.h` in `feature1.c`, cpplint will raise `should include its header file . Relative paths like . and .. are not allowed.  [build/include]`. 
These two errors don't make sense as my root directory should be the subdirectory.
